lib/mailchimp/updateNewsletterSubscription.js

```diff
@@ -0,0 +1,61 @@
+const fetch = require('isomorphic-unfetch')
+const logger = console
+const {
+  NewsletterSubscription,
+  getMemberApiUrl,
+  isEligibleForInterestId,
+  nameToInterestId
+} = require('./utils')
+
+module.exports = async ({ pgdb, userId, email, name, subscribed, roles }) => {
```

Comment:
For more clarity and reusability I would change the method signature to:
`module.exports = async ({ userId, newsletterName, subscribed }, context)`
then get pgdb and me from context `const { user: me, pgdb } = context`
then check for permissions with `Roles.userIsMeOrInRoles(user, me, ['supporter, admin'])`